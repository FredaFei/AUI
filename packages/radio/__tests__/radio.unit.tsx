import * as renderer from 'react-test-renderer'
import React from 'react'
import {shallow, mount} from 'enzyme'

import Radio from '../radio'

describe('测试 Radio 以下行为', () => {
  it('Radio 存在', () => {
    const json = renderer.create(<Radio checked>apple</Radio>).toJSON()
    expect(json).toMatchSnapshot()
  })
  it('Radio 设置 checked', () => {
    const wrapper = mount(<Radio checked>apple</Radio>)
    expect(wrapper.props().checked).toEqual(true)
    expect(wrapper.find('.am-radio-wrapper').hasClass('am-radio-checked')).toEqual(true)
  })
  it('Radio 设置 disabled', () => {
    const wrapper = mount(<Radio disabled>apple</Radio>)
    expect(wrapper.props().disabled).toEqual(true)
    expect(wrapper.find('.am-radio-wrapper').hasClass('am-radio-disabled')).toEqual(true)
  })
  it('Radio 设置 onChange', () => {
    const onChange = jest.fn()
    const wrapper = mount(<Radio checked={false} onChange={onChange}>apple</Radio>)
    wrapper.find('.am-radio').simulate('change')
    expect(onChange).toHaveBeenCalled();
    wrapper.setProps({checked:true});
    expect(wrapper.props().checked).toEqual(true)
    expect(wrapper.find('.am-radio-wrapper').hasClass('am-radio-checked')).toEqual(true)
  })
  it('Radio 设置 自定义类名', () => {
    const wrapper = shallow(<Radio className="test">apple</Radio>)
    expect(wrapper.find('.am-radio-wrapper').hasClass('test')).toBe(true)
  })
  it('Radio 设置 自定义样式', () => {
    const wrapper = shallow(<Radio style={{marginRight: '30px'}}>apple</Radio>)
    expect(wrapper.render()).toMatchSnapshot()
  });
})