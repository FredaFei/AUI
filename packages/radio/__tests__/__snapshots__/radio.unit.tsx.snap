// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 Radio 以下行为 Radio 存在 1`] = `
<label
  className="am-radio-wrapper am-radio-checked "
>
  <span
    className="am-radio-input"
  >
    <input
      className="am-radio"
      type="radio"
    />
  </span>
  <span
    className="am-radio-label"
  >
    apple
  </span>
</label>
`;

exports[`测试 Radio 以下行为 Radio 设置 自定义样式 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "am-radio-wrapper ",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "am-radio-input",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "am-radio",
              "style": "margin-right:30px",
              "type": "radio",
            },
            "children": Array [],
            "name": "input",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "style": undefined,
              "type": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "style": undefined,
              "type": undefined,
            },
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "am-radio-label",
          },
          "children": Array [
            Object {
              "data": "apple",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "am-radio-label",
        },
        "children": Array [
          Object {
            "data": "apple",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "am-radio-input",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "am-radio",
                "style": "margin-right:30px",
                "type": "radio",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "style": undefined,
                "type": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "style": undefined,
                "type": undefined,
              },
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "label",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
